Add documentation on how to use `USING`

talgalili opened the pull request:
(4th attempt at this point, combining github online/desktop and terminal did not go well for me...)

highker commented on presto-docs/src/main/sphinx/sql/select.rst:
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
+
+.. code-block:: none 
+
+    SELECT *
```
Could you use tpch tables as examples like other examples do?

talgalili replied:
I'm not sure what you mean here with "tpch tables". Can you help direct me to what it is you are referring to? (As far as I can see, other places in the doc uses VALUES for examples that can be ran, and the pseudo code I'm using at the beginning of the section, I feel, is better than a real example)

highker replied:
You could start `TpchQueryRunner` and use the tables in side that catalog.
After that you can query against tpch tables
```
cd presto/presto-cli/target
presto-cli-*-executable.jar --catalog tpch --schema sf10
```

All the examples we use are from tpch; so users can directly copy and paste a query on that webpage and run it from cli.

talgalili replied:
Got it.
Thanks for the suggestion, but I suspect this would only make the example more cumbersome. 
The first examples I wrote are not to be run at the terminal, but this trick has been used throughout the text (e.g.: in the "WITH Clause" section), and is (IMHO) a better way to explain the structure.

The last code I added is self contained (and based on VALUES) and the user can run it in the terminal to see how it behaves (this was also used in the text, e.g.: the sections with union, intersect, UNNEST, etc.).

In short, I don't think adding another example would be helpful to the text (and I think this diff is probably good to go as is).

If you have a specific example you think would be helpful, please share it so that I'd better understand what you had in mind.

What do you think?

highker commented on presto-docs/src/main/sphinx/sql/select.rst:
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
+
+For example:
+
+.. code-block:: none 
```
This is actually a sql code block. Use `::` after "For example"

highker commented on presto-docs/src/main/sphinx/sql/select.rst:
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
+    JOIN table_2
+    ON table_1.key_A = table_2.key_A AND table_1.key_B = table_2.key_B
+
+Can be rewritten to:
```
"Can" -> "can"

highker commented on presto-docs/src/main/sphinx/sql/select.rst:
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
+
+The output of doing ``JOIN`` with ``USING`` will be one copy of the join key 
+columns (key_1 and key_2 in the example above) followed by the remaining columns
+in table_1 and then the remaining columns in table_2. Note that the join keys are not
```
use quotes around example names (e.g., \`\`table_1\`\` or \`\`key_1\`\`)

highker commented on presto-docs/src/main/sphinx/sql/select.rst:
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
+
+Can be rewritten to:
+
+.. code-block:: none 
```
Same ::

highker commented on presto-docs/src/main/sphinx/sql/select.rst:
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
+
+The following two queries are equivalent:
+
+.. code-block:: none 
```
Same ::

talgalili commented:
Hi @highker,
Thanks for the comments!
I've implemented all of your suggestions (except one, which I didn't fully understand - please see my comment there).
I've squashed all the commits into one commit.
If this is good with you, feel free to pull/rebase/merge. If you want me to do more on the diff, please let me know.

Cheers,
T

highker commented on presto-docs/src/main/sphinx/sql/select.rst:
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
+
+The following two queries are equivalent::
+
+.. code-block:: none 
```
remove

highker commented on presto-docs/src/main/sphinx/sql/select.rst:
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
+
+can be rewritten to::
+
+.. code-block:: none 
```
remove

highker commented on presto-docs/src/main/sphinx/sql/select.rst:
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
+
+For example::
+
+.. code-block:: none 
```
remove

highker commented on presto-docs/src/main/sphinx/sql/select.rst:
```diff
@@ -804,6 +804,82 @@ The following query will fail with the error ``Column 'name' is ambiguous``::
+
+
+The output of doing ``JOIN`` with ``USING`` will be one copy of the join key 
+columns (``key_1`` and ``key_2`` in the example above) followed by the remaining columns
```
typo: `key_A`/`key_B`

talgalili replied:
Good catch, thanks :)

talgalili commented:
@highker - thanks for the comments, I've implemented all of them, committed, and squashed. I believe this diff is now ready to be rebased/merged.

Cheers,
T

highker commented:
Thanks @gal for your contribution!
